index.js

```diff
@@ -0,0 +1,100 @@
+#!/usr/bin/env node
+require("dotenv").config();
+const axios = require("axios");
+
+// Store Slack access token in a .env file in the root folder
+const accessToken = process.env.SLACK_ACCESS_TOKEN;
+
+async function getConversationId(conversation) {
+
+  // Trim the pound sign if it was provided in the conversation argument
+  if (conversation.startsWith('#')) {
+    conversation = conversation.slice(1)
+  }
+
+  // Destructuring to pull out the channels result from the response
+  const { data: { channels } } = await axios.get(`https://slack.com/api/conversations.list?token=${accessToken}&pretty=1`)
+
+  // Find the channel with matching conversation name and pull out its ID
+  const conversationId = channels.find(chan => chan.name === conversation).id
+
+  return conversationId
+}
+
+async function getConversationMembers(conversationId) {
+
+  async function queryForMembers(membersCollection = [], nextCursor = '') {
+    let allConversationMembers = [];
+
+    // Base case: members exist, no next cursor, we've retrieved all the members
+    if (membersCollection.length > 0 && !nextCursor) {
+      return membersCollection;
+    }
+
+    try {
+      const response = await axios.get(
+        `https://slack.com/api/conversations.members?token=${accessToken}&channel=${conversationId}${nextCursor}`
+      );
+
+      // Looks ugly, but pulling out members and next_cursor
+      let {
+        data: {
+          members,
+          response_metadata: { next_cursor }
+        }
+      } = response;
+
```

Comment:
I'm honestly not familiar with the above syntax.

Reply:
Assuming you are referring to the "Looks ugly, but..." bit, that's nested destructuring, pulling the results from the response object that axios returned. It could be rewritten as:

`let members = response.data.members`
`let next_cursor = response.data.response_metadata.next_cursor`

OR, with normal destructuring instead of nested:

`let { members } = response.data`
`let { next_cursor } = response.data.response_metadata`

Destructuring allows you to pull out a property from an object into a variable of the same name.

Reply:
@bhurstGH you should write this in a manner that is most clear to your audience. If Nate can't read it, you may want to consider changing it to the clearer version instead, or argue for it as the standard way to do things.

Given that you say in a comment: "Looks ugly..." It's probably not the clearest way to do things.

Reply:
We'll make improvements for the Slack App migration.